packages/server/src/queries/complex/portfolio/allocation.ts

```diff
@@ -0,0 +1,152 @@
+import { getAsset } from "../assets";
+
+interface FormattedAllocation {
+  key: string;
+  percentage: RatePretty;
+  fiatValue: PricePretty;
+  asset?: CoinPretty;
+}
+
+export interface GetAllocationResponse {
+  all: FormattedAllocation[];
+  assets: FormattedAllocation[];
+  available: FormattedAllocation[];
+}
+
+export function getAll(categories: Categories): FormattedAllocation[] {
+  const userBalancesCap = new Dec(categories["user-balances"].capitalization);
+  const stakedCap = new Dec(categories["staked"].capitalization);
+  const unstakingCap = new Dec(categories["unstaking"].capitalization);
+  const unclaimedRewardsCap = new Dec(
+    categories["unclaimed-rewards"].capitalization
+  );
+  const pooledCap = new Dec(categories["pooled"].capitalization);
+
+  const totalCap = userBalancesCap
+    .add(stakedCap)
+    .add(unstakingCap)
+    .add(unclaimedRewardsCap)
+    .add(pooledCap);
+
+  return [
+    {
+      key: "available",
+      percentage: new RatePretty(userBalancesCap.quo(totalCap)),
+      fiatValue: new PricePretty(DEFAULT_VS_CURRENCY, userBalancesCap),
+    },
+    {
+      key: "staked",
+      percentage: new RatePretty(stakedCap.quo(totalCap)),
+      fiatValue: new PricePretty(DEFAULT_VS_CURRENCY, stakedCap),
+    },
+    {
+      key: "unstaking",
+      percentage: new RatePretty(unstakingCap.quo(totalCap)),
+      fiatValue: new PricePretty(DEFAULT_VS_CURRENCY, unstakingCap),
+    },
+    {
+      key: "unclaimedRewards",
+      percentage: new RatePretty(unclaimedRewardsCap.quo(totalCap)),
+      fiatValue: new PricePretty(DEFAULT_VS_CURRENCY, unclaimedRewardsCap),
+    },
+    {
+      key: "pooled",
+      percentage: new RatePretty(pooledCap.quo(totalCap)),
+      fiatValue: new PricePretty(DEFAULT_VS_CURRENCY, pooledCap),
+    },
+  ];
+}
+
+export function calculatePercentAndFiatValues(
+  categories: Categories,
+  assetLists: AssetList[],
+  category: "total-assets" | "user-balances"
+) {
+  const totalAssets = categories[category];
+  const totalCap = new Dec(totalAssets.capitalization);
+
+  // Get top 5 assets by cap value
+  const sortedAccountCoinsResults =
+    totalAssets?.account_coins_result?.sort(
+      (a: AccountCoinsResult, b: AccountCoinsResult) =>
+        +b.cap_value - +a.cap_value
+    ) || [];
+
+  const top5AccountCoinsResults = sortedAccountCoinsResults.slice(0, 5);
+
+  const assets: FormattedAllocation[] = top5AccountCoinsResults.map(
+    (asset: AccountCoinsResult) => {
+      const assetFromAssetLists = getAsset({
+        assetLists,
+        anyDenom: asset.coin.denom,
+      });
+
+      return {
+        key: assetFromAssetLists.coinDenom,
+        percentage: new RatePretty(new Dec(asset.cap_value).quo(totalCap)),
+        fiatValue: new PricePretty(
+          DEFAULT_VS_CURRENCY,
+          new Dec(asset.cap_value)
+        ),
+      };
+    }
+  );
+
+  const otherAssets = sortedAccountCoinsResults.slice(5);
```

Comment:
NIT: let's try to avoid hardcoding with magic numbers. Thoughts on creating a number parameter on function perhaps called allocationLimit or something, that is defaulted to 5? Then it could be passed to both slices